### PR TITLE
Branch for testing DemoCards.jl

### PR DESCRIPTION
**Pull request by BeastyBlacksmith**
Also changes CI from travis to gh-actions

### CODE REVIEW

**Review by BeastyBlacksmith — COMMENTED** [review]
@johnnychen94 I got around to give DemoCards.jl a go. 

This is a mostly working version for GR alone, the titles look funny sometimes. Like `Open&#x2F;High&#x2F;Low&#x2F;Close`

It would be nice to have different backends as tabs.

File: docs/make.jl [hunk 1]
```diff
@@ -57,38 +60,44 @@ const PAGES = Any[
...
+#     generate_markdown(be)
+# end
 ansicolor = get(ENV, "PLOTDOCS_ANSICOLOR", "true") == "true"
 @show ansicolor
 @time makedocs(
     format = Documenter.HTML(
         prettyurls = get(ENV, "CI", nothing) == "true",
-        assets = ["assets/favicon.ico"],
+        assets = ["assets/favicon.ico",
+            # gallery_assets...
```
[the remark this answers is not in the document]
Comment (BeastyBlacksmith) [reply docs/make.jl:88]: seems its iterating the string

**Comment by BeastyBlacksmith** [comment]
`rm("docs/gallery/generated", recursive = true); generate_cards(:pgfplotsx); preview_demos("docs/gallery")` runs longer than I am willing to wait

**Comment by johnnychen94** [comment]
I'm glad that you have worked it out a few backends.

Sorry it's near the end of this semester and I can't comment much right now; I'll continue the work on DemoCards in Jan.

**Comment by BeastyBlacksmith** [comment]
This works now somewhat, but somehow all card links give a `ERR_FILE_NOT_FOUND`

**Comment by BeastyBlacksmith** [comment]
@johnnychen94 if I run `make.jl` I get the following error
```julia
ERROR: IOError: readdir("~/.julia/dev/PlotDocs.jl/docs/src/democards"): no such file or directory (ENOENT)
Stacktrace:
 [1] uv_error
   @ ./libuv.jl:97 [inlined]
 [2] readdir(dir::String; join::Bool, sort::Bool)
   @ Base.Filesystem ./file.jl:851
 [3] readdir
   @ ./file.jl:846 [inlined]
 [4] (::DemoCards.var"#107#111"{String, String, String, String, String, String, String, Vector{String}})()
   @ DemoCards ~/.julia/packages/DemoCards/w8V2k/src/generate.jl:214
 [5] (::var"#5#6")(cb::DemoCards.var"#107#111"{String, String, String, String, String, String, String, Vector{String}})
   @ Main ~/.julia/dev/PlotDocs.jl/docs/make.jl:103
 [6] foreach(f::var"#5#6", itr::Vector{Any})
   @ Base ./abstractarray.jl:2694
 [7] top-level scope
   @ ~/.julia/dev/PlotDocs.jl/docs/make.jl:101
```

**Comment by johnnychen94** [comment]
This might due to the fact that all demo files are generated from another function `generate_cards`. 

```julia
foreach(galleries_cb) do cb
    # URL redirection for DemoCards-generated gallery
    cb()
end
```

The callbacks are currently specific to redirect the URL of "Edit on Github" button. A workaround is to not use the generated callbacks and manually remove this button in the generated HTML files.

**Comment by BeastyBlacksmith** [comment]
preview is up again: https://docs.juliaplots.org/previews/PR256/
I don't get, why I get the titles for skipped examples for GR but not for the other backends.

**Comment by BeastyBlacksmith** [comment]
Also I have trouble getting that custom stylesheet working, it always complains about having an absolute path, though I did specify a relative path

**Comment by BeastyBlacksmith** [comment]
> Would you mind pointing me to which case is broken? I'm not sure what is expected and what is not.

On the [GR](https://docs.juliaplots.org/previews/PR256/generated_gr/) cards it shows e.g. "DataFrames" and the blank card, which is desired and on the [PlotlyJS](https://docs.juliaplots.org/previews/PR256/generated_plotlyjs/) cards its "plotlyjs_ref25" and same for the other backends.

**Comment by johnnychen94** [comment]
It seems everything works now.

From a user's perspective, I still dislike the empty cards; it feels like something is broken.

**Comment by BeastyBlacksmith** [comment]
> From a user's perspective, I still dislike the empty cards; it feels like something is broken.

That's actually the point. To show what is working on which backend and what is not. This lets you quickly evaluate the expectation about the functionality of a backend by scrolling through the galllery.

**Pull request merged**